### PR TITLE
[Ansible] Fixes #27 #28 # 29 #30 rpi config / donkeycar / openCV  role 

### LINKED ISSUE #27: [Ansible] Role "raspi-config", basic raspberry pi configurations
**Needs #26 to be merged before.**

Create a base role, this role will be used for generic raspberry pi configuration stuff not directly related to DonkeyCar projet.

This role should : 

- [x] Enable SSH, by creating /boot/ssh file, using Ansible [file module with state: touch](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/file_module.html)
- [x] Configure wifi, based on a template file that will be copied to `/etc/wpa_supplicant/wpa_supplicant.conf`. Parameters (country code, wifi ssid and psk) should use variables defined in group_vars/all
- [x] Perform update and upgrade using the [Ansible apt module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_module.html#ansible-collections-ansible-builtin-apt-module) see in the exemple task `Run the equivalent of "apt-get update" as a separate step`, for upgrade use `upgrade: safe` option see exemples in documentation
- [x] Change password for pi user, with password defined as variable, you might take a [look at this article](https://jianjye.medium.com/how-to-update-user-password-with-ansible-f971f41a3b3e) or simply use command `chpasswd` (see [how raspi-config does the job](https://github.com/raspberrypi-ui/rc_gui/blob/master/src/rc_gui.c#L132))
- [x] Enable I2C [look at here](https://github.com/raspberrypi-ui/rc_gui/blob/master/src/rc_gui.c#L81)
- [x] Enable camera

Take care to use named tasks and templates.

That's all :)

### PR DESCRIPTION
**Pull request by Benvii**
Fixes #27 #28 #29 #30

Add rpi-config related stuff role.

### CODE REVIEW

**Comment by yannis-mlgrn** [comment]
`%d` == 1
error : `FAILED! => {"msg": "Timeout (12s) waiting for privilege escalation prompt: "}`
solve by adding '-c paramiko' in command line

File: ansible/donkeycar.yml [hunk 1]
```diff
@@ -6,5 +6,9 @@
   remote_user: pi
 
   roles:
-    - role: shutdown-btn
+    - role: raspi-config #rasperry pi configuration role
       become: yes # This role need root privileges
+    - role: donkeycar # Donkeycar installation 
+      become: no # This role need root privileges
```
[the remark this answers is not in the document]
Comment (yannis-mlgrn) [reply ansible/donkeycar.yml:12]: resolve here [7a72438](https://github.com/mdl29/donkeycarLPH/pull/32/commits/7a72438c1682d08fc30a89faf2b0ae0130144466)

File: ansible/hosts [hunk 2]
```diff
@@ -4,7 +4,7 @@ monkecar.local
 bluecar.local
 
 [dev]
-raspberrypi.local ansible_ssh_common_args='-o StrictHostKeyChecking=no' ansible_user=pi ansible_password=raspberry
+raspberry.local ansible_ssh_common_args='-o StrictHostKeyChecking=no' ansible_user=pi ansible_password=raspberry
```
[the remark this answers is not in the document]
Comment (yannis-mlgrn) [reply ansible/hosts]: resolve here [7a72438](https://github.com/mdl29/donkeycarLPH/pull/32/commits/7a72438c1682d08fc30a89faf2b0ae0130144466)

File: ansible/donkeycar.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yannis-mlgrn) [reply ansible/donkeycar.yml]: Fixes here [6db0950](https://github.com/mdl29/donkeycarLPH/pull/32/commits/6db09504096cefcb56a1127feeb10fc6ab35d3f3)

**Comment by viandoxdev** [comment]
Aren't most of the vars defined in `group_vars/all` redundant with the ones defined in `roles/raspi-config/defaults/main.yml` ?

**Comment by Benvii** [comment]
Build tested at commit 1d8191b7c2e813d276aaf0cdc2fbfefde575913b , got the following error with packer : 
```
    arm-image.raspbian: TASK [donkeycar : Add virtualenv activation in bashrc file for pi user] ********
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [openCV : Install openCV dependencies] ************************************
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [openCV : Install openCV] *************************************************
    arm-image.raspbian: fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "No package matching 'opencv-python' is available"}
    arm-image.raspbian:
    arm-image.raspbian: PLAY RECAP *********************************************************************
    arm-image.raspbian: 127.0.0.1                  : ok=14   changed=13   unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    arm-image.raspbian:
==> arm-image.raspbian: fuser -k /tmp/armimg-014744649
Build 'arm-image.raspbian' errored after 1 hour 10 minutes: Error executing Ansible: Non-zero exit status: 2

==> Wait completed after 1 hour 10 minutes

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 2

==> Builds finished but no artifacts were created.
```

**Comment by Benvii** [comment]
Tested at commit a7408c33273cc025a1772ae3b99aca8aa868b246, failed with logs : 
```
==> arm-image.raspbian: Provisioning with Ansible...
    arm-image.raspbian: Uploading Playbook directory to Ansible staging directory...
    arm-image.raspbian: Creating directory: /tmp/packer-provisioner-ansible-local/625a0284-c4e3-1a0f-354f-df7ea005525e
    arm-image.raspbian: Uploading main Playbook file...
    arm-image.raspbian: Uploading inventory file...
    arm-image.raspbian: Executing Ansible: cd /tmp/packer-provisioner-ansible-local/625a0284-c4e3-1a0f-354f-df7ea005525e && /tmp/ansible/venv/bin/ansible-playbook /tmp/packer-provisioner-ansible-local/625a0284-c4e3-1a0f-354f-df7ea005525e/donkeycar.yml --extra-vars "packer_build_name=raspbian packer_builder_type=arm-image packer_http_addr=ERR_HTTP_ADDR_NOT_IMPLEMENTED_BY_BUILDER -o IdentitiesOnly=yes" --extra-vars ansible_python_interpreter=/tmp/ansible/venv/bin/python -c local -i /tmp/packer-provisioner-ansible-local/625a0284-c4e3-1a0f-354f-df7ea005525e/packer-provisioner-ansible-local2273796302
==> arm-image.raspbian: [DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the
==> arm-image.raspbian: controller starting with Ansible 2.12. Current version: 3.7.3 (default, Jan 22
==> arm-image.raspbian: 2021, 20:04:44) [GCC 8.3.0]. This feature will be removed from ansible-core in
==> arm-image.raspbian: version 2.12. Deprecation warnings can be disabled by setting
==> arm-image.raspbian: deprecation_warnings=False in ansible.cfg.
==> arm-image.raspbian: ERROR! no module/action detected in task.
==> arm-image.raspbian:
==> arm-image.raspbian: The error appears to be in '/tmp/packer-provisioner-ansible-local/625a0284-c4e3-1a0f-354f-df7ea005525e/roles/raspi-config/tasks/main.yml': line 32, column 3, but may
==> arm-image.raspbian: be elsewhere in the file depending on the exact syntax problem.
==> arm-image.raspbian:
==> arm-image.raspbian: The offending line appears to be:
==> arm-image.raspbian:
==> arm-image.raspbian:
==> arm-image.raspbian: - name: Enable pigpio
==> arm-image.raspbian:   ^ here
==> arm-image.raspbian: fuser -k /tmp/armimg-328912166
Build 'arm-image.raspbian' errored after 3 minutes 47 seconds: Error executing Ansible: Non-zero exit status: 4

==> Wait completed after 3 minutes 47 seconds

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 4

==> Builds finished but no artifacts were created.
```

**Comment by Benvii** [comment]
Fixed by a059470f5523e41770edb219c0b741b710a74f73, now got this error : 
```
    arm-image.raspbian: Executing Ansible: cd /tmp/packer-provisioner-ansible-local/625a07bf-8e72-5ebf-f6c4-07806f89d6c8 && /tmp/ansible/venv/bin/ansible-playbook /tmp/packer-provisioner-ansible-local/625a07bf-8e72-5ebf-f6c4-07806f89d6c8/donkeycar.yml --extra-vars "packer_build_name=raspbian packer_builder_type=arm-image packer_http_addr=ERR_HTTP_ADDR_NOT_IMPLEMENTED_BY_BUILDER -o IdentitiesOnly=yes" --extra-vars ansible_python_interpreter=/tmp/ansible/venv/bin/python -c local -i /tmp/packer-provisioner-ansible-local/625a07bf-8e72-5ebf-f6c4-07806f89d6c8/packer-provisioner-ansible-local2722252991
==> arm-image.raspbian: [DEPRECATION WARNING]: Ansible will require Python 3.8 or newer on the
==> arm-image.raspbian: controller starting with Ansible 2.12. Current version: 3.7.3 (default, Jan 22
==> arm-image.raspbian: 2021, 20:04:44) [GCC 8.3.0]. This feature will be removed from ansible-core in
==> arm-image.raspbian: version 2.12. Deprecation warnings can be disabled by setting
==> arm-image.raspbian: deprecation_warnings=False in ansible.cfg.
==> arm-image.raspbian: ERROR! conflicting action statements: chdir, ansible.builtin.command
==> arm-image.raspbian:
==> arm-image.raspbian: The error appears to be in '/tmp/packer-provisioner-ansible-local/625a07bf-8e72-5ebf-f6c4-07806f89d6c8/roles/mycar/tasks/main.yml': line 3, column 3, but may
==> arm-image.raspbian: be elsewhere in the file depending on the exact syntax problem.
==> arm-image.raspbian:
==> arm-image.raspbian: The offending line appears to be:
==> arm-image.raspbian:
==> arm-image.raspbian:
==> arm-image.raspbian: - name: create car
==> arm-image.raspbian:   ^ here
==> arm-image.raspbian: fuser -k /tmp/armimg-499041280
Build 'arm-image.raspbian' errored after 4 minutes 5 seconds: Error executing Ansible: Non-zero exit status: 4

==> Wait completed after 4 minutes 5 seconds

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 4

==> Builds finished but no artifacts were created.
```

**Comment by Benvii** [comment]
I finally managed to relaunch it for the night ... let's see the result tomorrow hope it will not take several days to validated it. Please do not update this branch anymore it needs to be stabilize otherwise nothing will be merge, we should work tickets by tickets, small steps.

**Comment by yannis-mlgrn** [comment]
> Aren't most of the vars defined in `group_vars/all` redundant with the ones defined in `roles/raspi-config/defaults/main.yml` ?

Yes because in group_vars we put the var of the client, he can change if he want. And in defaults folder we put the default variables of the installation if he put nothing in the group_vars/all

**Comment by Benvii** [comment]
Now go this erreur, venv isn't sourced : 
```
    arm-image.raspbian: TASK [shutdown-btn : Copy systemd service file to server] **********************
    arm-image.raspbian: changed: [127.0.0.1]
    arm-image.raspbian:
    arm-image.raspbian: TASK [mycar : create car] ******************************************************
    arm-image.raspbian: fatal: [127.0.0.1]: FAILED! => {"changed": true, "cmd": "cd ~\ndonkey createcar --path ~/mycar\n", "delta": "0:00:00.028529", "end": "2022-04-16 02:27:02.520584", "msg": "non-zero return code", "rc": 127, "start": "2022-04-16 02:27:02.492055", "stderr": "/bin/sh: 2: donkey: not found", "stderr_lines": ["/bin/sh: 2: donkey: not found"], "stdout": "", "stdout_lines": []}
    arm-image.raspbian:
    arm-image.raspbian: RUNNING HANDLER [raspi-config : Start pigpio service] **************************
    arm-image.raspbian:
    arm-image.raspbian: RUNNING HANDLER [shutdown-btn : Start button-shutdown service] *****************
    arm-image.raspbian:
    arm-image.raspbian: PLAY RECAP *********************************************************************
    arm-image.raspbian: 127.0.0.1                  : ok=19   changed=18   unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    arm-image.raspbian:
==> arm-image.raspbian: fuser -k /tmp/armimg-566433678
Build 'arm-image.raspbian' errored after 1 hour 12 minutes: Error executing Ansible: Non-zero exit status: 2

==> Wait completed after 1 hour 12 minutes

==> Some builds didn't complete successfully and had errors:
--> arm-image.raspbian: Error executing Ansible: Non-zero exit status: 2

==> Builds finished but no artifacts were created.
```

**Pull request merged**